File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.py
def solution(n, wires):
    answer = n
    for i in range(n-1):
        tmp = wires.copy()
        tmp.pop(i)
        
        graph = [[] for _ in range(n + 1)]
        for a, b in tmp:
            graph[a].append(b)
            graph[b].append(a)
        visited = [False]*(n+1)
        q = [1]
        visited[1] = True
        cnt = 1
        while q:
            now = q.pop(0)
            for d in graph[now]:
                if visited[d]: continue
                visited[d] = True
                q.append(d)
                cnt += 1
        tmp = abs(cnt-(n-cnt))
        answer = min(tmp, answer)
        
        
    return answer

File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.js
function solution(n, wires) {
  var answer = n;

  for (let i = 0; i < wires.length; i++) {
    const tmp = wires.slice();
    tmp.splice(i, 1);

    const graph = Array.from({ length: n + 1 }, () => []);
    for (const [a, b] of tmp) {
      graph[a].push(b);
      graph[b].push(a);
    }

    const visited = Array(n + 1).fill(false);
    const q = [1];
    visited[1] = true;
    let cnt = 1;

    while (q.length) {
      const now = q.shift();
      for (const i of graph[now]) {
        if (visited[i]) continue;
        visited[i] = true;
        q.push(i);
        cnt++;
      }
    }

    let num = Math.abs(cnt - (n - cnt));
    answer = Math.min(answer, num);
  }

  return answer;
}
